shell/models/fleet.cattle.io.gitrepo.js

```diff
@@ -339,9 +344,10 @@ export default class GitRepo extends SteveModel {
           namespacedName = `${ r.namespace }:${ r.name }`;
         }
 
-        let state = r.state;
+        let state = r.state; // FIXME: why assign r.state when it's always overwritten?
+
         const perEntry = r.perClusterState?.find((x) => x.clusterId === c.id);
-        const tooMany = r.perClusterState?.length >= 10 || false;
+        const tooMany = r.perClusterState?.length >= 10 || false; // FIXME: shouldn't we not calc perEntry if there are too many?
```

Comment:
Additionally, Fleet creates an incomplete flag on the  resource and shouldn't add more than 10 perCluster states.